napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -396,15 +469,40 @@ def _get_dialog(self) -> QDialog:
             p = p.parent()
         return p
 
-    def set_busy(self, text: str, update: bool = False):
+    def set_busy(
+        self, text: str, action_name: str = None, update: bool = False
+    ):
+        """Updates status text and what buttons are visible when any button is pushed.
+
+        Parameters
+        ----------
+        text: str
+            The new string to be displayed as the status.
+        action_name: str
+            The action of the button pressed.
+        update: bool
+            States whether this install is an update or not.
+
+        """
         self.item_status.setText(text)
-        self.cancel_btn.setVisible(True)
-        if not update:
+        if action_name == 'install' and update is True:
+            self.cancel_btn.setVisible(True)
             self.action_button.setVisible(False)
-        else:
-            self.update_btn.setVisible(False)
+            self.old_action = 'update'
```

Comment:
What is the purpose of `self.old_action`? It seems to be set here but never queried. Apart from setting this attribute, this `if` and its first `elif` are identical so I think we could simplify here.